src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProject.cs

```diff
@@ -239,6 +241,27 @@ public ParseOptions ParseOptions
                        w => w.OnParseOptionsChanged(Id, value));
         }
 
+        internal string CommandLineOptions
+        {
+            get => _commandLineOptions;
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException(nameof(value));
+                }
+
+                if (_activeBatchScopes > 0)
```

Comment:
Shouldn't;t this be `_activeBatchScopes == 0`?

Reply:
Actually doesn't `ChangeProjectProperty`  handle the batch scope check?

---
In reply to: [252487160](https://github.com/dotnet/roslyn/pull/32977#discussion_r252487160) [](ancestors = 252487160)

Reply:
Ah it actually does.